### ISSUE TITLE
Track llama_*_embeddings_nextn bindings required for MTP

### ISSUE THREAD

**Issue by ardan-bkennedy** [hybridgroup/yzma#344]
@deadprogram, this is a follow-up to #253 with an update on both the observed stability and the remaining upstream caveat.

The APIs are **not yet officially stable public C APIs**. They still live in `src/llama-ext.h`, which explicitly labels itself a staging/WIP header where breaking changes and C++ are allowed. That header is not included in llama.cpp's installed public headers, and its declarations are outside `extern "C"`, so current shared libraries may expose C++-mangled symbols even though the declarations use `LLAMA_API`.

That said, the canonical names and signatures have now remained unchanged since ggml-org/llama.cpp commit [`166fe29492ab`](https://github.com/ggml-org/llama.cpp/commit/166fe29492ab) on June 3, 2026:

```cpp
LLAMA_API void    llama_set_embeddings_nextn(struct llama_context * ctx, bool value, bool masked);
LLAMA_API float * llama_get_embeddings_nextn(struct llama_context * ctx);
LLAMA_API float * llama_get_embeddings_nextn_ith(struct llama_context * ctx, int32_t i);
```

They are still identical in [b10930](https://github.com/ggml-org/llama.cpp/blob/b10930/src/llama-ext.h#L94-L108), 1,435 upstream commits later. Kronk has also exercised them through a local FFI extension across multiple llama.cpp updates without a signature change. This is empirical stability, not an upstream ABI guarantee.

These functions are required to implement MTP speculative decoding correctly. An MTP head is not a standalone language model: each draft step needs both the token ID and the target model's hidden state. Kronk uses the APIs to:

1. Call `llama_set_embeddings_nextn(ctx_tgt, true, false)` so the target context retains every output row.
2. Call `llama_set_embeddings_nextn(ctx_dft, true, true)` so the draft context retains only logits-selected rows.
3. Read the target hidden-state buffer with `llama_get_embeddings_nextn` and individual output rows with `llama_get_embeddings_nextn_ith`.
4. Shift/copy those hidden states into `batch.embd` for the MTP draft decode, matching llama.cpp's `common/speculative.cpp` implementation.

Without access to these functions, Yzma consumers cannot implement the upstream MTP data flow; Kronk must dynamically load both canonical and mangled symbol names and disables MTP when they are unavailable.

This issue is intended to track Yzma support for the three functions—either once llama.cpp promotes them to its regular C API, or earlier if Yzma is willing to bind the staging interface given its demonstrated stability. Once Yzma provides them, Kronk can remove its local FFI workaround.
